// chatbot-widget/dist/ChatbotUI.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import React, { useState, useEffect, useRef } from 'react';
import { rasaApi } from './services/rasaApi';
export default function ChatbotUI({ avatarUrl }) {
    const [messages, setMessages] = useState([
        { from: 'bot', text: 'Ciao! Come posso aiutarti oggi?' },
    ]);
    const [input, setInput] = useState('');
    const [isOpen, setIsOpen] = useState(false);
    const [showPreview, setShowPreview] = useState(false);
    const [sendMessage] = rasaApi.endpoints.sendMessage.useMutation();
    const messagesEndRef = useRef(null);
    const defaultAvatar = avatarUrl || 'https://via.placeholder.com/40';
    const handleSend = (userText, payload) => __awaiter(this, void 0, void 0, function* () {
        const trimmed = userText.trim();
        if (!trimmed)
            return;
        setMessages((prev) => [...prev, { from: 'user', text: trimmed }]);
        setInput('');
        try {
            const data = yield sendMessage({
                sender: 'utente123',
                message: payload || trimmed,
            }).unwrap();
            const botMessages = data.map((res) => ({
                from: 'bot',
                text: res.text || '',
                buttons: res.buttons || undefined,
            }));
            setMessages((prev) => [...prev, ...botMessages]);
        }
        catch (err) {
            console.error('Errore:', err);
            setMessages((prev) => [
                ...prev,
                { from: 'bot', text: 'Errore nel contattare il server.' },
            ]);
        }
    });
    useEffect(() => {
        var _a;
        (_a = messagesEndRef.current) === null || _a === void 0 ? void 0 : _a.scrollIntoView({ behavior: 'smooth' });
    }, [messages]);
    useEffect(() => {
        const timer = setTimeout(() => {
            if (!isOpen)
                setShowPreview(true);
        }, 3000);
        return () => clearTimeout(timer);
    }, [isOpen]);
    return (React.createElement(React.Fragment, null,
        !isOpen && (React.createElement(React.Fragment, null,
            React.createElement("button", { className: "chatbot-float", onClick: () => {
                    setIsOpen(true);
                    setShowPreview(false);
                } },
                React.createElement("img", { src: defaultAvatar, alt: "Apri Chatbot" })),
            showPreview && (React.createElement("div", { className: "chatbot-preview-bubble" }, messages[0].text)))),
        isOpen && (React.createElement("div", { className: "chatbot-wrapper" },
            React.createElement("div", { className: "chatbot-head" },
                React.createElement("div", { className: "chatbot-head-left" },
                    React.createElement("img", { src: defaultAvatar, alt: "Avatar", className: "chatbot-avatar" }),
                    React.createElement("span", null, "Chatbot Biblioteca")),
                React.createElement("span", { className: "chatbot-toggle", onClick: () => setIsOpen(false) }, "\u00D7")),
            React.createElement("div", { className: "chatbot-box" },
                messages.map((msg, i) => (React.createElement("div", { key: i, className: `chatbot-message ${msg.from}` },
                    React.createElement("div", { className: `chatbot-bubble ${msg.from}` }, msg.text),
                    msg.buttons && (React.createElement("div", { className: "chatbot-buttons" }, msg.buttons.map((btn, idx) => (React.createElement("button", { key: idx, onClick: () => handleSend(btn.title, btn.payload), className: "chatbot-button" }, btn.title)))))))),
                React.createElement("div", { ref: messagesEndRef })),
            React.createElement("input", { value: input, onChange: (e) => setInput(e.target.value), onKeyDown: (e) => e.key === 'Enter' && handleSend(input), placeholder: "Scrivi il tuo messaggio...", className: "chatbot-input" })))));
}
